src/include/miopen/performance_config.hpp

```diff
@@ -68,6 +68,23 @@ struct PerfConfigBase : PerfConfig
     }
 };
 
+template <class Derived>
+struct PerfConfigBaseCK : PerfConfig
+{
+    void Serialize(std::ostream& stream) const final
+    {
+        const Derived& self = static_cast<const Derived&>(*this);
+        stream.write(self.kernel_id.c_str(), self.kernel_id.length());
+    }
+
+    bool Deserialize(const std::string& s) final
+    {
+        Derived& self  = static_cast<Derived&>(*this);
+        self.kernel_id = s;
```

Comment:
[R] The `Derived& self...` stuff looks cryptic to me. I recommend more clear definition, like
```C++
auto instance = static_cast<Derived&>(*this)
instance.kernel_id = ...
```
or just inlining the cast.

Reply:
I prefer the code the way it is when compared to using `auto` (as too much of auto hurts readability). 
In reality, this case does indicate a broken interface. I discussed with @CAHEK7 on how we should have this base class `PerfConfigBaseCK` and move the members like `kernel_id` up to this common base instead of have 4 or 5 derived classes, each with the same member `kernel_id`.

Reply:
So in short, I recommend moving `kernel_id` up to this base class. I or @CAHEK7 can take it up as he's already dealt with the derived classes recently.

Reply:
CK base class is a kind of tricky solution and it's still in progress.